pages/badge-single.jsx

```diff
@@ -0,0 +1,217 @@
+                    </div>
+
+
+
+
+                    <img src="/img/pages/badges/svg/divider.svg" alt="" className="center-block horizontal-divider"/>
+                </div>
+
+
+
+
+
+                <div className="badge-requirement">
+                    <h3 className={'text-light'}>Badge Requirements</h3>
+                    <p>Make or write something that demonstrates your understanding of any two or more of the following:</p>
+
+                    <RequirementsList
+                        list={this.state.requirements}
+                        icon="fa fa-check"/>
+                </div>
+
+
+                <div className="apply-send-qualifications">
+                    <p><strong className={'text-bold'}>Ideas?</strong> You could submit a blog post, a project you made using Mozilla's tools, or another web creation you made. Demonstrate your understanding in your own unique way!</p>
+
+
+                    <h3 className={'text-light'}>Apply for this badge</h3>
+
+                    <form className="horizontal-form" role="form" onSubmit={this.onQualificationsSubmit}>
+                        <div className="form-group">
+                            <label htmlFor="qualifications" className="control-label">What qualifies you to earn this badge?</label>
+                            <textarea
+                                name="qualifications"
+                                ref="qualifications"
+                                id="qualifications"
+                                cols="30"
+                                rows="10"
+                                className="form-control"
+                                placeholder="I've earned this badge by working on this project: [link]. My project demonstrates an understanding of [skill/competency] through [explain further]."></textarea>
+                        </div>
+
+                        <div className="optional-file-input">
+                            <input type="file" className="hidden" name="optional_file" id="optional_file" ref="optionalFile" />
+                            <button type="button" ref="optionalFileBtn" className="btn btn-link" onClick={this.handleFileSelect}>Add Optional Attachment(s)</button>
+                        </div>
+
+                        <div>
+                            <button type="submit" className="btn btn-awsm">Apply</button>
+                        </div>
+                    </form>
+
+                </div>
+
+                <img src="/img/pages/badges/svg/divider.svg" alt="" className="center-block horizontal-divider"/>
+
+                <div className="navigation">
+                    <SingleBadgePageNavigation
+                        nextNavigationDetails={this.state.navigation.next}
+                        prevNavigationDetails={this.state.navigation.prev} />
+                </div>
+
+            </div>
+        );
+    }
+});
+
+module.exports = BadgePage;
```

Comment:
this file can probably do with a bit of whitespace cleanup (but with a newline at the end of the file so github doesn't flag it as problematic)